src/services/layerHelper.js

```diff
@@ -0,0 +1,316 @@
+goog.provide('ngeo.LayerHelper');
+
+goog.require('ngeo');
+goog.require('ol.Collection');
+goog.require('ol.format.WMTSCapabilities');
+goog.require('ol.layer.Group');
+goog.require('ol.layer.Image');
+goog.require('ol.layer.Tile');
+goog.require('ol.source.ImageWMS');
+goog.require('ol.source.WMTS');
+
+
+
+/**
+ * Provides helpers fonctions that helps you to create and manage layers.
+ *
+ * @param {angular.$q} $q Angular promises/deferred service.
+ * @param {angular.$http} $http Angular http service.
+ * @constructor
+ * @ngdoc service
+ * @ngname ngeoLayerHelper
+ * @ngInject
+ */
+ngeo.LayerHelper = function($q, $http) {
+
+  /**
+   * @type {boolean}
+   * @private
+   */
+  this.initialized_ = false;
+
+  /**
+   * @type {ol.Map}
+   * @private
+   */
+  this.map_ = null;
+
+  /**
+   * @type {angular.$q}
+   * @private
+   */
+  this.$q_ = $q;
+
+  /**
+   * @type {angular.$http}
+   * @private
+   */
+  this.$http_ = $http;
+
+  /**
+   * @type {string}
+   * @export
+   */
+  this.helperID = 'helperID';
+};
+
+
+/**
+ * Init the service.
+ * @param {!ol.Map} map
+ * @export
+ */
+ngeo.LayerHelper.prototype.init = function(map) {
```

Comment:
If I add a "map" parameters at two fonctions (``addLayerToMap`` and ``removeLayerFromMap``, that seems not too bad to do that) It can manage multiple maps. Wait...